virttest/qemu_vm.py

```diff
@@ -3828,22 +3820,23 @@ def migrate(self, timeout=virt_vm.BaseVM.MIGRATE_TIMEOUT, protocol="tcp",
                 'qemu_binary'] = utils_misc.get_qemu_dst_binary(self.params)
         if env:
             env.register_vm("%s_clone" % clone.name, clone)
-        if (local and not (migration_exec_cmd_src and
-                           "gzip" in migration_exec_cmd_src)):
-            error_context.context("creating destination VM")
-            if stable_check:
-                # Pause the dest vm after creation
-                extra_params = clone.params.get("extra_params", "") + " -S"
-                clone.params["extra_params"] = extra_params
-
-            clone.create(migration_mode=protocol, mac_source=self,
-                         migration_fd=fd_dst,
-                         migration_exec_cmd=migration_exec_cmd_dst)
-            if fd_dst:
-                os.close(fd_dst)
-            error_context.context()
 
         try:
+            if (local and not (migration_exec_cmd_src and
+                               "gzip" in migration_exec_cmd_src)):
+                error_context.context("creating destination VM")
+                if stable_check:
+                    # Pause the dest vm after creation
+                    extra_params = clone.params.get("extra_params", "") + " -S"
+                    clone.params["extra_params"] = extra_params
+
+                clone.create(migration_mode=protocol, mac_source=self,
```

Comment:
When destination vm creation failed, it skips to line 3989 and execute code in line 3994. It'll send "cont" to the source vm which is in "running" status already. So a better solution is preferred.

Reply:
@aliang123 What is the better solution for this? When destination vm creation failed, the test result will be error or fail. It will be destroyed later, no matter source vm is "paused" or "running",

Reply:
@aliang123  I can understand your point. I think that is for the destination vm when there is no errors.

Reply:
@hereischen, suggest to check vm status after line 3993, if vm.status==paused, then execute line 3994.

Reply:
@aliang123 updated, thanks.